www/js/lib/uiUtil.js

```diff
@@ -45,35 +45,72 @@ define(rqDef, function(settingsStore) {
-            document.getElementById("approveConfirm").style.display = isConfirm ? "inline" : "none";
-            document.getElementById("declineConfirm").style.display = isConfirm ? "inline" : "none";
-            document.getElementById("closeMessage").style.display = isConfirm ? "none" : "inline";
+            document.getElementById('approveConfirm').style.display = isConfirm ? 'inline' : 'none';
+            document.getElementById('declineConfirm').style.display = isConfirm ? 'inline' : 'none';
+            document.getElementById('closeMessage').style.display = isConfirm ? 'none' : 'inline';
             // Display the modal
-            $("#alertModal").modal("show");
-            // When hide model is called, resolve promise with true if hidden using approve button, false otherwise
-            $("#alertModal").on("hide.bs.modal", function () {
-                const closeSource = document.activeElement;
-                if (closeSource.id === "approveConfirm") {
-                    resolve(true);
-                } else {
-                    resolve(false);
-                }
-            });
-            document.getElementById('alertModal').addEventListener('keyup', function (e) {
+            const modal = document.querySelector('#alertModal');
+            const backdrop = document.createElement('div');
+            backdrop.classList.add('modal-backdrop');
+            document.body.appendChild(backdrop);
+
+            // Show the modal
+            document.body.classList.add('modal-open');
+            modal.classList.add('show');
+            modal.style.display = 'block';
+            backdrop.classList.add('show');
+
+            // Set the ARIA attributes for the modal
+            modal.setAttribute('aria-hidden', 'false');
+            modal.setAttribute('aria-modal', 'true');
+            modal.setAttribute('role', 'dialog');
+
+            // Hide modal handlers
+            function closeModalHandler(){
+              document.body.classList.remove('modal-open');
+              modal.classList.remove('show');
+              modal.style.display = 'none';
+              backdrop.classList.remove('show');
+              if(Array.from(document.body.children).indexOf(backdrop)>=0){ 
+                document.body.removeChild(backdrop);
+              }
+              //remove event listeners
+              document.getElementById('modalCloseBtn').removeEventListener('click', close);
+              document.getElementById('declineConfirm').removeEventListener('click', close);
+              document.getElementById('closeMessage').removeEventListener('click', close);
+              document.getElementById('approveConfirm').removeEventListener('click', closeConfirm);
+              modal.removeEventListener('click', close);
+              document.getElementsByClassName('modal-dialog')[0].removeEventListener('click', stopOutsideModalClick);
+              modal.removeEventListener('keyup', keyHandler);
```

Comment:
For consistency, please use four-space indentation for all of the statements in this function, not two-space indentation.